prompt2model/utils/api_tools.py

```diff
@@ -179,33 +174,23 @@ async def _throttled_completion_acreate(
         return responses
 
 
-def handle_api_error(e, api_call_counter=0):
+def handle_api_error(e) -> None:
     """Handle OpenAI errors or related errors that the API may raise.
 
     Args:
         e: The error to handle. This could be an OpenAI error or a related
            non-fatal error, such as JSONDecodeError or AssertionError.
-        api_call_counter: The number of API calls made so far.
-
-    Returns:
-        The captured exception (if error is API related and can be retried),
-        else raise the error.
     """
     logging.error(e)
+    if not isinstance(e, API_ERRORS):
+        raise e
```

Comment:
With this change we'll no longer be logging the errors where we suppress->retry. Without this change the log will have something like:
```
ERROR:root:Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
```

Is this what you were intending?